src/testutils.rs

```diff
@@ -81,16 +81,11 @@ impl Token {
         let from_id = to_ed25519(&self.env, from);
         let nonce = self.nonce(&from_id);
 
-        let mut args: Vec<RawVal> = Vec::new(&self.env);
-        args.push_back(from_id.clone().into_val(&self.env));
-        args.push_back(nonce.clone().into_val(&self.env));
-        args.push_back(to.clone().into_val(&self.env));
-        args.push_back(amount.clone().into_val(&self.env));
         let msg = SignaturePayload::V0(SignaturePayloadV0 {
             function: symbol!("xfer"),
             contract: self.contract_id.clone(),
             network: self.env.ledger().network_passphrase(),
-            args,
+            args: (from_id.clone(), nonce.clone(), to.clone(), amount.clone()).into_val(&self.env),
```

Comment:
Fyi should be possible to use refs too.

Reply:
Oh nice let me try that. I need to open another PR anyways because I forgot to update `approve`.